core/base/v7/inc/ROOT/RIndexIter.hxx

```diff
@@ -32,13 +32,20 @@ namespace Internal {
 
 template <class REFERENCE,
           class POINTER = typename std::add_pointer<typename std::remove_reference<REFERENCE>::type>::type>
-class RIndexIter: public std::iterator<std::random_access_iterator_tag, REFERENCE, POINTER> {
+class RIndexIter {
```

Comment:
@pcanal Could you please take a look. I simply "mocked" the current behaviour, but I suspect is wrong. The iterator was given as:
```cpp
template <class REFERENCE,
          class POINTER = typename std::add_pointer<typename std::remove_reference<REFERENCE>::type>::type>
class RIndexIter: public std::iterator<std::random_access_iterator_tag, REFERENCE, POINTER>
```
which implied that the `value_type` is of type `REFERENCE` and the `difference_type` is of type `POINTER`. Was that intentional? Or the difference is a `REFERENCE` as well?

Reply:
In my mind, `REFERENCE` should be a reference and `POINTER` as pointer like:
```
   using iterator_category = std::random_access_iterator_tag;
   using value_type = typename std::remove_reference<REFERENCE>::type;
   using difference_type = std::ptrdiff_t;
   using pointer = POINTER;
   using reference = REFERENCE;
```
But it is good question for @Axel-Naumann.

Reply:
Just looking how `RIndexIter` used later in `RHistBinIter.hxx`

Reply:
I agree it's weird. It's also likely to vanish once @hahnjo finds the time to re-implement views (and with it likely iteration). So I propose you just convert it 1:1 as you did, thanks!